### PR TITLE
Fix broken `Evaluate Lua` funtionality in Lua Debugger

### LINKED ISSUE #7734: Debugger sometimes locks up the editor completely
**Describe the bug (REQUIRED)**
In some scenarios, the editor might lock up completely when interacting with the debugger. It seems there is an issue with the debugger.

**To Reproduce (REQUIRED)**
1. Launch a game and attach a debugger
2. Hit pause
3. Evaluate something using the "Evaluate Lua" text field. No response is received.
4. Hit resume

**Expected behavior (REQUIRED)**
The editor does not lock up. Also, the evaluation results are displayed in the output.

**Defold version (REQUIRED):**
Latest

**Platforms (REQUIRED):**
Desktop

### PR DESCRIPTION
**Pull request by AGulev**
Fix issue when Lua Debugger hangs after using `Evaluate Lua` functionality.

Fix https://github.com/defold/defold/issues/7734

## PR checklist

* [x] Code
	* [ ] Add engine and/or editor unit tests.
	* [x] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [x] Prepare pull request and affected issue for automatic release notes generator
	* [x] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [x] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [x] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [x] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

### CODE REVIEW

File: engine/engine/content/builtins/scripts/mobdebug.lua [hunk 1]
```diff
@@ -592,16 +592,21 @@ local function get_server_data(file)
     return
   end
   checkcount = mobdebug.checkcount
-  if coroutine.status(coro_debugger) == "suspended" then
-    local status, res = cororesume(coro_debugger, nil,nil,file)
+  if corostatus(coro_debugger) == "suspended" then
+    local state_vars = capture_vars(3)
+    local status, res = cororesume(coro_debugger, nil, state_vars, file)
```
Comment (AGulev) [inline engine/engine/content/builtins/scripts/mobdebug.lua:597]: Current state should be taken to set it as environment to be able to Evaluate Lua code
But default `vars` should be overwritten to be able to restore environment of the debugger itself.

**Review by vlaaad — APPROVED** [review]
LGTM! I tried this new mobdebug script, and it no longer locks up the editor, and evaluation results are now received by the editor!

One note, this debugger still responds to `SUSPEND` command with this:
```
401 Error in Execution 3
[c]
```
Previously, it responded with something like this:
```
202 Paused builtins/render/default.render_script 40
```

Still, it's much better now and the editor no longer locks up, ship it!

File: engine/engine/content/builtins/scripts/mobdebug.lua (hunk 1 above)
Comment (britzl) [reply engine/engine/content/builtins/scripts/mobdebug.lua:597]: What's the difference between corostatus and coroutine.status?

Comment (AGulev) [reply engine/engine/content/builtins/scripts/mobdebug.lua:597]: It's just a mobdebug wrapper for coroutine.status(): https://github.com/defold/defold/blob/de77e1c99f36a56e64594bc93d6098580dc471ee/engine/engine/content/builtins/scripts/mobdebug.lua#L65

**Review by britzl — CHANGES_REQUESTED** [review]
Question about corostatus vs coroutine.status

**Review by britzl — APPROVED** [review]

**Pull request merged**